apps/passport-client/src/dispatch.ts

```diff
@@ -216,7 +236,46 @@ async function genDeviceLoginPassport(identity: Identity, update: ZuUpdate) {
   });
 }
 
-async function login(
+async function createNewUserSkipPassword(
+  email: string,
+  token: string,
+  state: AppState,
+  update: ZuUpdate
+) {
+  update({
+    modal: { modalType: "none" }
+  });
+  const crypto = await PCDCrypto.newInstance();
+  const encryptionKey = await crypto.generateRandomKey();
+  await saveEncryptionKey(encryptionKey);
+
+  update({
+    encryptionKey
+  });
+
+  const newUserResult = await requestCreateNewUser(
+    appConfig.zupassServer,
+    email,
+    token,
+    state.identity.commitment.toString(),
+    undefined,
+    encryptionKey
+  );
+
+  if (newUserResult.success) {
+    return finishLogin(newUserResult.value, state, update);
+  }
+
+  update({
+    error: {
+      title: "Account creation failed",
+      message: "Couldn't create an account. " + newUserResult.error,
+      dismissToCurrentPage: true
+    }
+  });
+}
+
```

Comment:
it's important to be deliberate and careful about the order in which you update state and perform long-running actions in dispatch - updating state causes a re-render, which will have an opportunity to cause react to be in a sort of intermidate state that you didn't account for in the case that you didn't think through it carefully.